openmdao/docs/theory_manual/total_derivs/total_derivs_theory.rst

```diff
@@ -0,0 +1,227 @@
+Second if the model includes nonlinear solvers, then noise in the nonlinear convergence for very small steps also introduces error.
+
+To achieve better accuracy, the `complex-step <http://mdolab.engin.umich.edu/content/complex-step-derivative-approximation-0>`_ method can be used:
+
+.. math::
+
+    \frac{d f}{d x} = \frac{\text{Im}[f(x+ i h)]}{h} + \mathcal{O}(h^2)
+
+This offers second order convergence with the size of the step in the complex plain :math:`h`, but more importantly avoids the issue with subtractive cancellation so that :math:`h` can be chosen to be arbitrarily small and then essentially perfect accuracy can be achieved.
+The improved accuracy does come at the expense of greater computational cost and implementation complexity.
+In most cases changes are required to the underlying model, including any nonlinear solvers, to handle the complex arithmetic which is roughly 2.5 times more expensive than basic real arithmetic.
+
+
+Basic Approach for Analytic Total Derivatives
+----------------------------------------------
+
+While the numerical approach to computing total derivatives is conceptually simpler, analytic approaches are more accurate and less computationally expensive.
+The analytic approaches all follow a two step recipe for computing total derivatives.
+
+    #. directly compute the partial derivatives for each component
+    #. solve a linear system to compute the total derivatives
+
+Partial derivatives are simpler and less costly to computing directly, as compared to total derivatives, because in the context of OpenMDAO they only involve a single component and never require re-converging of the nonlinear analysis.
+OpenMDAO offers several different ways to compute partial derivatives,
+but for now we'll assume that the partials are simply known quantities and focus on how to execute step 2 of the process above.
+
+Consider a function :math:`f=F(x,y)` where :math:`x` is a design variable vector of length :math:`n` and :math:`y` is an implicit variable vector of length :math:`m`.
+Any evaluation of :math:`F` must include the convergence of the nonlinear system :math:`R(x,y)=0`.
+We can expand the total derivative as follows:
+
+.. math::
+    \frac{d f}{d x} = \frac{\partial F}{\partial x} + \frac{\partial F}{\partial y}\frac{d y}{d x}
+
+
+:math:`\partial F/\partial x` and :math:`\partial F/\partial y` are partial derivative terms, which we have assumed are known quantities.
+We need a way to compute :math:`d y/d x`, which represents the change in the converged values of :math:`y` for a change in :math:`x`.
+There are two methods available to compute this term: direct and adjoint.
+
+Direct Method
+***************
+
+The direct method can be derived by recognizing that since :math:`R(x,y)=0` when the model is converged, and we can differentiate the residual function :math:`R` with respect to :math:`x`:
```

Comment:
the "since" in this sentence is essentially an if. so the "and we can differentiate" should really be "THEN we can differentiate."